test/e2e/clustertask/start_test.go

```diff
@@ -0,0 +1,119 @@
+import (
+	"testing"
+
+	"github.com/AlecAivazis/survey/v2/terminal"
+	"github.com/Netflix/go-expect"
+	"github.com/tektoncd/cli/test/e2e"
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+	"gotest.tools/v3/icmd"
+	knativetest "knative.dev/pkg/test"
+)
+
+const (
+	tePipelineGitResourceName = "skaffold-git"
+)
+
+func TestClusterTaskInteractiveStartE2E(t *testing.T) {
+	t.Parallel()
+	c, namespace := e2e.Setup(t)
+	knativetest.CleanupOnInterrupt(func() { e2e.TearDown(t, c, namespace) }, t.Logf)
+	defer e2e.TearDown(t, c, namespace)
+
+	kubectl := e2e.NewKubectl(namespace)
+	tkn, err := e2e.NewTknRunner(namespace)
+	if err != nil {
+		t.Fatalf("Error creating tknRunner %+v", err)
+	}
+
+	t.Logf("Creating clustertask ")
+	e2e.Assert(t, kubectl.Create(e2e.ResourcePath("read-file-clustertask.yaml")), icmd.Success)
+
+	t.Logf("Creating git pipeline resource in namespace: %s", namespace)
+	e2e.Assert(t, kubectl.Create(e2e.ResourcePath("git-resource.yaml")), icmd.Success)
+
+	t.Run("Get list of ClusterTasks", func(t *testing.T) {
+		res := tkn.Run("clustertask", "list")
+		expected := e2e.ListAllClusterTasksOutput(t, c, map[int]interface{}{
+			0: &e2e.TaskData{
+				Name: "read-task",
+			},
+		})
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+			Out:      expected,
+		})
+	})
+
+	t.Run("Start ClusterTask with flags", func(t *testing.T) {
+		res := tkn.Run("clustertask", "start", "read-task",
+			"-i=source="+tePipelineGitResourceName,
+			"-p=FILENAME=docs/README.md",
+			"--showlog",
+			"true")
+
+		vars := make(map[string]interface{})
+		taskRunGeneratedName := e2e.GetTaskRunListWithName(c, "read-task").Items[0].Name
+		vars["Taskrun"] = taskRunGeneratedName
+		expected := e2e.ProcessString(`(TaskRun started: {{.Taskrun}}
+Waiting for logs to be available...
+.*)`, vars)
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+		})
+		assert.Assert(t, is.Regexp(expected, res.Stdout()))
+	})
+
+	t.Run("Start ClusterTask interactively", func(t *testing.T) {
+		tkn.RunInteractiveTests(t, &e2e.Prompt{
+			CmdArgs: []string{"clustertask", "start", "read-task"},
+			Procedure: func(c *expect.Console) error {
+				if _, err := c.ExpectString("Choose the git resource to use for source:"); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("skaffold-git (https://github.com/GoogleContainerTools/skaffold)"); err != nil {
+					return err
+				}
+
+				if _, err := c.SendLine(string(terminal.KeyEnter)); err != nil {
+					return err
+				}
+				if _, err := c.ExpectString("Value for param `FILENAME` of type `string`?"); err != nil {
+					return err
+				}
+				if _, err := c.SendLine("docs/README.md"); err != nil {
+					return err
+				}
+				if _, err := c.ExpectEOF(); err != nil {
+					return err
+				}
+
+				c.Close()
+				return nil
+			}})
+	})
+
+	t.Logf("Deleting clustertask ")
+	e2e.Assert(t, kubectl.Delete(e2e.ResourcePath("read-file-clustertask.yaml")), icmd.Success)
```

Comment:
Deleting the clustertask is unnecessary. It will be cleaned up after the tests complete.